routes/index.js

```diff
@@ -38,22 +45,121 @@ router.get('/login', function (req, res, next) {
+    });
+});
+
+router.get('/add-to-cart/breakfast/:id', function (req, res, next) {
+    let bfItemId = req.params.id;
+    let cart = new Cart(req.session.cart ? req.session.cart : {});
+    Breakfast.findById(bfItemId, function (err, bfItem) {
+        if(err){
+            return res.redirect('/', {title: 'Something went wrong.'});
+        }
+        cart.add(bfItem, bfItem.id);
+        req.session.cart = cart;
+        console.log(req.session.cart);
+        res.redirect('/breakfast');
     });
 });
 
 //get lunch page
 router.get('/lunch', function (req, res, next) {
     Lunch.find(function (err, docs) {
-        res.render('lunch', {title:'Lunch Menu', lunch: docs});
+        res.render('menus/lunch', {title:'Lunch Menu', lunch: docs});
+    });
+});
+
+router.get('/add-to-cart/lunch/:id', function (req, res, next) {
+    let lunchItemId = req.params.id;
+    let cart = new Cart(req.session.cart ? req.session.cart : {});
+    Lunch.findById(lunchItemId, function (err, lunchItem) {
+        if(err){
+            return res.redirect('/', {title: 'Something went wrong.'});
+        }
+        cart.add(lunchItem, lunchItem.id);
+        req.session.cart = cart;
+        console.log(req.session.cart);
+        res.redirect('/lunch');
     });
 });
 
 //get dinner menu
 router.get('/dinner', function (req, res, next) {
     Dinner.find(function (err, docs) {
-        res.render('dinner', {title:'Dinner Menu', dinner:docs});
+        res.render('menus/dinner', {title:'Dinner Menu', dinner:docs});
     });
 });
 
+router.get('/add-to-cart/dinner/:id', function (req, res, next) {
+    let dinnerItemId = req.params.id;
+    let cart = new Cart(req.session.cart ? req.session.cart : {});
+    Dinner.findById(dinnerItemId, function (err, dinnerItem) {
+        if(err){
+            return res.redirect('/', {title: 'Something went wrong.'});
+        }
+        cart.add(dinnerItem, dinnerItem.id);
+        req.session.cart = cart;
+        console.log(req.session.cart);
+        res.redirect('/dinner');
+    });
+});
+
+router.get('/shopping-cart', function (req, res, next) {
+    if(!req.session.cart){
+        return res.render('shopping-cart',{foodItems: null});
+    }
+    let cart = new Cart(req.session.cart);
+    res.render('shopping-cart', {foodItems: cart.generateArray(), totalPrice: cart.totalPrice})
+});
+
+router.get('/checkout', isLoggedIn, function (req, res, next) {
+    if(!req.session.cart){
+        return res.redirect('/shopping-cart');
+    }
+    let cart = new Cart(req.session.cart);
+    res.render('checkout',{total: cart.totalPrice});
+});
+
+router.post('/checkout', isLoggedIn, function (req, res, next) {
+    if(!req.session.cart){
+        return res.redirect('/shopping-cart');
+    }
+    let cart = new Cart(req.session.cart);
+    let stripe = require("stripe")("sk_test_Nm4ul2p0g79wCgLhUB5xBZoD");
+
+    stripe.charges.create({
+        amount: cart.totalPrice * 100,
```

Comment:
Best Practice: Magic Number - A number with no apparent meaning but just makes things work within programming. This ```100``` is a magic number. I get that you are trying to move a decimal with it, but theres not a lot of explanation why. Consider a better semantic way to approach this. Here's a [bit more reading and an explanation](https://codeburst.io/software-anti-patterns-magic-numbers-7bc484f40544) of how to solve.

Reply:
created a issue for this one but haven't addressed it yet https://github.com/piazzaana/chicken-fries/issues/62 (magic number is funny lol), it was the way I figured things worked with stripe. I'll revise that

Reply:
Remember that tutorials always use bad variable names and dumb things down as much as possible. We want our code to be better 😄